### ISSUE TITLE
[Multi-user] Permission problems in package caching when mixing with other channels

### ISSUE THREAD

**Issue by frol** [conda/conda#3760]
I am using a multi-user setup when conda is installed into a read-only location and current user cannot write to it. Having this setup, I continue experience issues ever since Conda 4.1 release (the latest fine release was 4.0.9).

Currently, the issue I have stems from that [`conda.core.package_cache.find_new_location`](https://github.com/conda/conda/blob/27295f791cd8d86c6d4bebae3626f81047987c08/conda/core/package_cache.py#L143) finds `zlib` package, which is installed into a read-only location, and tries to download a new package from conda-forge into that read-only location:

```
$ conda create --yes -n _root --channel conda-forge zlib
...
    Traceback (most recent call last):
      File "/opt/conda/lib/python3.5/site-packages/conda/exceptions.py", line 479, in conda_exception_handler
        return_value = func(*args, **kwargs)
      File "/opt/conda/lib/python3.5/site-packages/conda/cli/main.py", line 145, in _main
        exit_code = args.func(args, p)
      File "/opt/conda/lib/python3.5/site-packages/conda/cli/main_create.py", line 68, in execute
        install(args, parser, 'create')
      File "/opt/conda/lib/python3.5/site-packages/conda/cli/install.py", line 422, in install
        raise CondaSystemExit('Exiting', e)
      File "/opt/conda/lib/python3.5/contextlib.py", line 77, in __exit__
        self.gen.throw(type, value, traceback)
      File "/opt/conda/lib/python3.5/site-packages/conda/cli/common.py", line 573, in json_progress_bars
        yield
      File "/opt/conda/lib/python3.5/site-packages/conda/cli/install.py", line 420, in install
        raise CondaRuntimeError('RuntimeError: %s' % e)
    conda.exceptions.CondaRuntimeError: Runtime error: RuntimeError: Runtime error: Could not open '/opt/conda/pkgs/zlib-1.2.8-3.tar.bz2.part' for writing ([Errno 13] Permission denied: '/opt/conda/pkgs/zlib-1.2.8-3.tar.bz2.part').
```

This issue is reproducible in Conda 4.1.12, 4.2.11, and current master.

Here is a Docker one-liner:

```
$ docker run --rm --user nobody continuumio/miniconda3 conda create --yes -n _root -c conda-forge zlib
```

/cc @jakirkham (I just figured that since it is related to Conda-Forge package clashing, you might be interested in this issue)

**Comment by frol** [comment]
Hi @HugoTian,

May I ask you to take a look into this bug? You have already fixed a related #3307, so it might be something you know how to fix easily...

**Comment by kalefranz** [comment]
multi-user installs are fixed with the latest conda 4.3
